project/MiMa.scala

```diff
@@ -0,0 +1,94 @@
+// It would be nice to use sbt-mima-plugin here, but the plugin is missing
+// at least two features we need:
+// * ability to run MiMa twice, swapping `curr` and `prev`, to detect
+//   both forwards and backwards incompatibilities
+// * ability to pass a filter file
+// So we invoke the MiMa CLI directly; it's also what the Ant build did.
+
+import sbt._
+import sbt.Keys._
+
+object MiMa {
+
+  lazy val mimaOldVersion =
+    settingKey[String]("Scala version number to run MiMa against")
+
+  lazy val mima =
+    taskKey[Unit]("run Migration Manager to detect binary incompatibilities")
+
+  lazy val settings =
+    Seq(
+      mimaOldVersion := VersionUtil.versionProps("scala.binary.version") + ".0",
```

Comment:
I assumed that MiMa would only be enabled after a .0 release. We can add a separate base version setting to the sbt build. It's also hardcoded in build.xml. The ant build reacts to a `test.bc.skip` setting but I don't know if it's actually used. I would change it in the CI script or build.sbt as required (especially if the base version is also set there; these two settings belong together)